includes/formatting.php

```diff
@@ -745,7 +746,16 @@ function give_clean($var)
         return array_map('give_clean', $var);
     }
 
-    return is_scalar($var) ? sanitize_text_field(wp_unslash($var)) : $var;
+    // remove serialized data
+    if (is_serialized($var)) {
+        return '';
+    }
+
+    if (is_scalar($var)) {
+        return sanitize_text_field(wp_unslash($var));
+    }
+
+    return '';
```

Comment:
I am nervous about this too, but this function shouldn't allow serialized data. We use this function to sanitize the user input, and if we are accepting serialized objects sent over HTTP, we are probably doing something wrong. 

The security report suggested that `give-form-title` is vulnerable to OI attack, so instead of changing the `give_clean` function, this [PR](https://github.com/impress-org/givewp/pull/7461/files) might be enough. But what about other places where we use `give_clean` to sanitize the user input, places like `LegacyPaymentGatewayRegisterAdapter`.

Reply:
@alaca that's a good point.  I guess where this would be a problem or have the most risk is if we were using this function for serialized data coming from the database.